core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -395,42 +396,51 @@ class GroupCoordinator(val brokerId: Int,
+    }
+  }
+
   def handleCommitOffsets(groupId: String,
                           memberId: String,
                           generationId: Int,
                           offsetMetadata: immutable.Map[TopicPartition, OffsetAndMetadata],
-                          responseCallback: immutable.Map[TopicPartition, Errors] => Unit,
-                          producerId: Long = RecordBatch.NO_PRODUCER_ID,
-                          producerEpoch: Short = RecordBatch.NO_PRODUCER_EPOCH) {
-    if (!isActive.get) {
-      responseCallback(offsetMetadata.mapValues(_ => Errors.COORDINATOR_NOT_AVAILABLE))
-    } else if (!isCoordinatorForGroup(groupId)) {
-      responseCallback(offsetMetadata.mapValues(_ => Errors.NOT_COORDINATOR))
-    } else if (isCoordinatorLoadInProgress(groupId)) {
-      responseCallback(offsetMetadata.mapValues(_ => Errors.COORDINATOR_LOAD_IN_PROGRESS))
-    } else {
-      groupManager.getGroup(groupId) match {
-        case None =>
-          if (generationId < 0) {
-            // the group is not relying on Kafka for group management, so allow the commit
-            val group = groupManager.addGroup(new GroupMetadata(groupId))
-            doCommitOffsets(group, memberId, generationId, producerId, producerEpoch, offsetMetadata, responseCallback)
-          } else {
-            // or this is a request coming from an older generation. either way, reject the commit
-            responseCallback(offsetMetadata.mapValues(_ => Errors.ILLEGAL_GENERATION))
-          }
+                          responseCallback: immutable.Map[TopicPartition, Errors] => Unit) {
+    validateGroup(groupId) match {
+      case Some(error) => responseCallback(offsetMetadata.mapValues(_ => error))
+      case None =>
+        groupManager.getGroup(groupId) match {
+          case None =>
+            if (generationId < 0) {
+              // the group is not relying on Kafka for group management, so allow the commit
+              val group = groupManager.addGroup(new GroupMetadata(groupId))
+              doCommitOffsets(group, memberId, generationId, NO_PRODUCER_ID, NO_PRODUCER_EPOCH,
+                offsetMetadata, responseCallback)
+            } else {
+              // or this is a request coming from an older generation. either way, reject the commit
+              responseCallback(offsetMetadata.mapValues(_ => Errors.ILLEGAL_GENERATION))
+            }
 
-        case Some(group) =>
-          doCommitOffsets(group, memberId, generationId, producerId, producerEpoch, offsetMetadata, responseCallback)
-      }
+          case Some(group) =>
+            doCommitOffsets(group, memberId, generationId, NO_PRODUCER_ID, NO_PRODUCER_EPOCH,
+              offsetMetadata, responseCallback)
+        }
     }
   }
 
   def handleTxnCompletion(producerId: Long,
-                          topicPartitions: Seq[TopicPartition],
+                          offsetsPartitions: Iterable[TopicPartition],
```

Comment:
Yes, I debated this a bit. It would be nice if we had a `Partitions` class or something to represent a set of partitions for a single topic. For now, perhaps I'll just assert the right topic.